app/controllers/controls_controller.rb

```diff
@@ -0,0 +1,51 @@
+class ControlsController < ApplicationController
+  def index
+    @controls = Control.order(:name)
+  end
+
+  def show
+    @control = Control.find(params[:id])
+  end
+
+  def new
+    @control = Control.new
+  end
+
+  def create
+    @control = Control.new(control_params)
+    if @control.save
+      redirect_to @control, notice: "Control created successfully"
+    else
+      render "new"
+    end
+  end
+
+  def edit
+    @control = Control.find(params[:id])
+  end
+
+  def update
+    @control = Control.find(params[:id])
+    if @control.update(control_params)
+      redirect_to @control, notice: "Control updated successfully"
+    else
+      render "edit"
+    end
+  end
+
+  def destroy
+    @control = Control.find(params[:id])
+
+    if @control.destroy
```

Comment:
You might want to go with `@control.destroy!` and drop the else since there isn't a user friendly reason this would fail

Reply:
You're right there isn't _yet_ – but once we get to the point where this interacts with VAIS, there's a non-zero chance the remote deletion fails for some transient reason so I figure we might as well.